File: Iniciante/URI 1047.js
var input = require("fs").readFileSync("/dev/stdin", "utf8");
var lines = input.split("\n");

let line = lines.shift().split(" ");

let hoursInitial = parseInt(line[0]);
let minuteInitial = parseInt(line[1]);
let hoursFinal = parseInt(line[2]);
let minuteFinal = parseInt(line[3]);

let minutes = 0;
let hours = 0;

if (
  hoursInitial == hoursFinal &&
  minuteInitial == minuteFinal &&
  hoursInitial == minuteFinal
) {
  console.log("O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)");
} else {
  if (hoursInitial > hoursFinal) {
    hours = 24 - hoursInitial + hoursFinal;
  } else if (hoursInitial < hoursFinal) {
    hours = 24 - (24 - hoursFinal + hoursInitial);
  }
  if (minuteInitial > minuteFinal) {
    minutes = 60 + minuteFinal - minuteInitial;
    if (hoursInitial == hoursFinal) {
      hours = 24 + hours - 1;
    } else {
      hours = hours - 1;
    }
  } else {
    minutes = minuteFinal - minuteInitial;
  }
  console.log(`O JOGO DUROU ${hours} HORA(S) E ${minutes} MINUTO(S)`);
}

File: Iniciante/URI 1047.pas
program uri_1047;
uses crt;

var h1,m1,h2,m2,ht,mt : integer;

BEGIN
	readln(h1,m1,h2,m2);
	ht:= h2 - h1;
	mt:= m2 - m1;
	
	if (mt < 0)
	then begin mt:= 60 + mt; ht:= ht-1 end;
	if (ht < 0)
	then ht := 24 + ht;
	if (h1 = h2) and (m1 = m2)
	then writeln('O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)')
	else writeln('O JOGO DUROU ',ht,' HORA(S) E ',mt,' MINUTO(S)')
	
	
END.
